### PR TITLE
Bug 1978798: Update ovnKubernetesConfig.policyAuditConfig Default

### PR DESCRIPTION
**Pull request by astoycos**
We need to initialize `ovnKubernetesConfig.policyAuditConfig`
so that the proper defaults are populated when a CNO config
object is created

Signed-off-by: astoycos <astoycos@redhat.com>

### CODE REVIEW

File: operator/v1/0000_70_cluster-network-operator_01_crd.yaml [hunk 1]
```diff
@@ -239,6 +239,7 @@ spec:
                         policyAuditConfig:
                           description: policyAuditConfig is the configuration for network policy audit events. If unset, reported defaults are used.
                           type: object
+                          default: null
```
Comment (sttts) [inline operator/v1/0000_70_cluster-network-operator_01_crd.yaml:242]: this looks wrong. What is kubebuilder doing here?

Comment (astoycos) [reply operator/v1/0000_70_cluster-network-operator_01_crd.yaml:242]: Not sure Maybe I need to specify the object here? Let me give that a try.

Comment (sttts) [reply operator/v1/0000_70_cluster-network-operator_01_crd.yaml:242]: Two ways to solve this:
1. fix kubebuilder's controller-gen upstream
2. use a yaml patch as we do in similar cases where kubebuilder is broken. Look in this repo for yaml-patch files as blueprint. A yaml patch allows you to set `{}` here explicitly as value. It is applied **after** controller-gen has been run.

Comment (astoycos) [reply operator/v1/0000_70_cluster-network-operator_01_crd.yaml:242]: +1 sounds good I'll give a stab at debugging kubebuilder first, thanks!

**Comment by astoycos** [comment]
@sttts I bashed my head against kubebuilder for a while trying to fix this, however based on the design of controller-tools it doesn't seem like a very trivial thing to fix.  I opened https://github.com/kubernetes-sigs/controller-tools/issues/622 and asked for help in [upstream](https://kubernetes.slack.com/archives/CAR30FCJZ/p1631031785123700)/ and [downstream](https://coreos.slack.com/archives/C3VS0LV41/p1631541185331800) slack channels to no avail. :( 

For now I'd like to get this bug fixed with the following patch and then fix properly upstream once I can get more help/time.

File: operator/v1/0000_70_cluster-network-operator_01_crd.yaml [hunk 2]
```diff
@@ -1,443 +0,0 @@
-apiVersion: apiextensions.k8s.io/v1
```
Comment (sttts) [inline operator/v1/0000_70_cluster-network-operator_01_crd.yaml:1]: please don't rename files and change them in the same commit. I cannot review this easily now.

File: operator/v1/0000_70_cluster-network-operator_01.crd.yaml-patch [hunk 3]
```diff
@@ -0,0 +1,7 @@
+- op: add
+  path: /spec/versions/name=v1/schema/openAPIV3Schema/properties/spec/properties/defaultNetwork/properties/ovnKubernetesConfig/properties/policyAuditConfig/default
+  value:
+    destination: "null"
```
Comment (sttts) [inline operator/v1/0000_70_cluster-network-operator_01.crd.yaml-patch:4]: no need to duplicate the per-field defaults. Just `{}` as value should be enough.

File: operator/v1/types_network.go [hunk 4]
```diff
@@ -372,6 +372,9 @@ type OVNKubernetesConfig struct {
 	IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
 	// policyAuditConfig is the configuration for network policy audit events. If unset,
 	// reported defaults are used.
+	// This config is replaced during a strategic merge patch
+	// TODO(astoycos): once https://github.com/kubernetes-sigs/controller-tools/issues/622
```
Comment (sttts) [inline operator/v1/types_network.go:376]: this ends up in the docs, probably not what you want.

File: operator/v1/types_network.go [hunk 5]
```diff
@@ -372,6 +372,9 @@ type OVNKubernetesConfig struct {
 	IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
 	// policyAuditConfig is the configuration for network policy audit events. If unset,
 	// reported defaults are used.
+	// This config is replaced during a strategic merge patch
```
Comment (sttts) [inline operator/v1/types_network.go:375]: there is no strategic merge patch for CRDs

**Comment by astoycos** [comment]
Sounds good I'll fix these... Also when I changed the naming to `.crd.yaml` from `_crd.yaml` we also seemed to have added some additional formatting somewhere to provide a maximum line length which wasn't done on the `_crd.yaml` this results in reformatting of the entire file so I'll add another commit for that too

**Comment by sttts** [comment]
The patch tool uses another library to format the yaml. So that's expected, ugly, but expected.

**Comment by astoycos** [comment]
ACK I though  I did something wrong :) 

Removed the comments, should be good to go

File: operator/v1/types_network.go (hunk 5 above)
Comment (astoycos) [reply operator/v1/types_network.go]: removed

File: operator/v1/types_network.go (hunk 4 above)
Comment (astoycos) [reply operator/v1/types_network.go]: removed

**Pull request merged**